src/distribution.cpp

```diff
@@ -19,32 +19,35 @@
 namespace openmc {
 
 //==============================================================================
-// Discrete implementation
+// DiscreteIndex implementation
 //==============================================================================
 
-Discrete::Discrete(pugi::xml_node node)
+DiscreteIndex::DiscreteIndex(pugi::xml_node node)
 {
   auto params = get_node_array<double>(node, "parameters");
+  std::size_t n = params.size() / 2;
 
-  std::size_t n = params.size();
-  std::vector<double> x_vec(params.begin(), params.begin() + n / 2);
-  std::vector<double> p_vec(params.begin() + n / 2, params.end());
+  prob_.assign(params.begin() + n, params.end());
 
-  this->init_alias(x_vec, p_vec);
+  this->init_alias();
 }
 
-Discrete::Discrete(const double* x, const double* p, int n)
+DiscreteIndex::DiscreteIndex(const double* p, int n)
 {
-  std::vector<double> x_vec(x, x + n);
-  std::vector<double> p_vec(p, p + n);
+  prob_.assign(p, p + n);
 
-  this->init_alias(x_vec, p_vec);
+  this->init_alias();
 }
 
-void Discrete::init_alias(vector<double>& x, vector<double>& p)
+void DiscreteIndex::assign(const double* p, int n)
+{
+  prob_.assign(p, p + n);
+
+  this->init_alias();
+}
```

Comment:
Did you mean to use this method in the two constructors above?

Reply:
I thought about it, but didn't want to think about how the `std::vector` iterator would be caste to `double *`.... but I've pushed it now.